src/donations.py

```
@@ -0,0 +1,93 @@
+"""
+Business logic for the donations feature.
+Validation, amount handling, and record formatting.
+Email notifications live in api/donations.py (runtime layer).
+"""
+import re
+from typing import Optional, Tuple
```

Ruff (0.15.18)

[warning] `typing.Tuple` is deprecated, use `tuple` instead

(UP035)